### ISSUE TITLE
3.9 and 3.8 are unstable - deploy spec is not valid in docs

### ISSUE THREAD

**Issue by chrisdlangton** [docker/compose#8490]
docker-compose version 1.29.2, build 5becea4c
Docker version 20.10.7, build f0df350
Linux pop-os 5.11.0-7620-generic #21~1624379747~21.04~3abeff8-Ubuntu SMP Wed Jun 23 02:34:03 UTC  x86_64 x86_64 x86_64 GNU/Linux

```yaml
version: '3.9'
services:
  redis:
    image: docker.io/library/redis:5.0
    deploy:
      restart_policy:
        condition: on-failure
        delay: 0s
        max_attempts: 10
        window: 3s
      resources:
        limits:
          cpus: '1.00'
          memory: 1GB
        reservations:
          cpus: '0.50'
          memory: 500M
```

produced

```
WARNING: The following deploy sub-keys are not supported and have been ignored: restart_policy.delay, restart_policy.window, resources.reservations.cpus
```

Tried to set the version from `3.9` (which is [the version in docs](https://docs.docker.com/compose/compose-file/compose-file-v3/#resources)) to `3.8` to match [the latest in the docs](https://docs.docker.com/compose/compose-file/compose-file-v3/) just in case the docs examples are prepared for an unreleassed version..

No change, same result

these docs, the cli output, i.e. the brittle experience of documented expectations and reality makes it impossible to know what the next troubleshooting step should be

**Comment by dwalisson** [comment]
Same here:

"The following deploy sub-keys are not supported and have been ignored: restart_policy.delay, restart_policy.window"

Docker version 20.10.10, build b485636
docker-compose version 1.29.2, build 5becea4c
Debian GNU/Linux bookworm/sid
Linux 5.14.0-4-amd64 #1 SMP Debian 5.14.16-1 (2021-11-03) x86_64 GNU/Linux

`version: '3.9'

services:
  nginx:
    container_name: nginx
    image: nginx:1.21.3-alpine
    deploy:
      resources:
        limits:
          cpus: 0.2
          memory: 512M
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s`

**Comment by nk9** [comment]
I'm seeing the same warning.

```
$ docker --version
Docker version 20.10.11, build dea9396
$ docker-compose --version
docker-compose version 1.29.2, build 5becea4c
$ sw_vers
ProductName:	macOS
ProductVersion:	11.6.2
BuildVersion:	20G314
```

**Comment by ndeloof** [comment]
tl;dr: You should adopt Compose V2.

The actual reference for compose file format is the Compose Specification (https://github.com/compose-spec/compose-spec), which makes use of `version` obsolete.

Legacy docker-compose version 1.29.* does not fully implement the compose specification, so this warning about unsupported keys (but, still, can process your compose file).

**Comment by nk9** [comment]
Thanks for the explanation. As of now, the documentation says that [1.29.2 is the current stable release](https://docs.docker.com/compose/install/). I didn't actually install docker-compse independently on my Mac, I just downloaded Docker Desktop and it came along.

Do you know when will version 2 become stable?

**Comment by ndeloof** [comment]
Compose v2 **is** stable, and distributed by Docker Desktop. Check on preferences you have "Use Docker Compose V2" enabled
